### ISSUE TITLE
translate-c: variable declaration in switch block produces invalid zig code

### ISSUE THREAD

**Issue by Pyrolistical** [ziglang/zig#14809]
### Zig Version

0.11.0-dev.1857+010596c93

### Steps to Reproduce and Observed Behavior

Given `repo.c`:
```c
static int foo (int x) {
  switch (x) {
    int y;
    case 0:
      y = 3;
      x = y;
      break;
  }
  return x;
}
```

Run:
```sh
zig translate-c repo.c > repo.zig
zig build-lib repo.zig 
```

Errors with:
```
repo.zig:60:17: error: use of undeclared identifier 'y'
                y = 3;
```

`repo.zig` at line 60:
```zig
pub fn foo(arg_x: c_int) callconv(.C) c_int {
    var x = arg_x;
    while (true) {
        switch (x) {
            @as(c_int, 0) => {
                y = 3;
                x = y;
                break;
            },
            else => {},
        }
        break;
    }
    return x;
}
```

### Expected Behavior

Expect repo lib to be built without error.

**Comment by Pyrolistical** [comment]
The workaround is to move `int y;` out of the switch block in repo.c, but that can change behaviour.

**Comment by Pyrolistical** [comment]
And the C code is valid.

![Screenshot 2023-03-05 102742](https://user-images.githubusercontent.com/463642/222978793-146a1a97-33a5-4ca3-920a-47901daf79d6.png)

I originally encountered this in sqlite. You can see a more real-world example by downloading https://www.sqlite.org/2023/sqlite-amalgamation-3410000.zip.

Search for `YYMINORTYPE yylhsminor;` in `sqlite.c`.

**Comment by Pyrolistical** [comment]
@vrischmann How does zig-sqlite avoid this issue?

**Comment by squeek502** [comment]
> How does zig-sqlite avoid this issue?

`zig-sqlite` doesn't use `translate-c` to compile `sqlite`, it just compiles the C files directly. `translate-c` is an unnecessary step if you're just trying to compile some C code (e.g. `zig build-lib repo.c` will work).

**Comment by rsepassi** [comment]
Yeah, I can confirm that cimport of sqlite.h and a regular build of the amalgamation works fine, but translate c breaks because of the switch issue reported here. The break applies to all uses of the Lemon parser generator that's part of the SQLite code base. Sometimes a translate c step is useful, for example if you want to call a C function at comptime, so it'd be nice for this to be fixed.
